exercises/bowling/bowling.exs

```diff
@@ -1,29 +1,30 @@
 defmodule Bowling do
+  @type t :: any()
```

Comment:
I don't see what this type definition adds here. What was your intention for adding this?

Reply:
To make clear that the previous `any`s were all the same data structure. Perhaps `@opaque` would be more appropriate? The type would still need to be specified as `any` since we leave that up the the exercist.

Reply:
I don't think we necessarily need to make it clear that they're the same data structure since they technically don't have to be to make the tests pass.

Also, `t()` is (in most cases other than a few special ones in Elixir itself like `String.t()` or `Keyword.t()`) reserved for the definition of a struct that is defined within a module. Other uses for defining type `t()` are exceedingly rare in real world Elixir.

Reply:
Perhaps `bowling_game()` then? I don't think `t()` is bad, though, since it suggests but does not require a struct.
The data structures _do_ need to be the same data structure, though, something made clear by reading the tests but not by the typespecs. The first arguments to `roll` and `score` must always be the structure returned by `start`.

Reply:
Technically they don't need to be the same. For example, there could be:
```elixir
def roll(game, roll) when is_list(game) do
  game
  |> Enum.into(%{})
  |> roll(roll)
end

def roll(game, roll) when is_map(game) do
  # implementation here that returns a keyword list
end
```